src/server.c

```diff
@@ -6992,6 +6992,12 @@ int main(int argc, char **argv) {
     char *exec_name = strrchr(argv[0], '/');
     if (exec_name == NULL) exec_name = argv[0];
     server.sentinel_mode = checkForSentinelMode(argc,argv, exec_name);
+    if (strstr(exec_name,"redis-server") != NULL) {
+        server.server_symlink = true;
+    } else {
+        server.server_symlink = false;
+    }
+
```

Comment:
`exec_name` is `argv[0]` and it's stored as `server.exec_argv[0]` some 15 lines below this.

Reply:
removed introduced new check and updated the diff with exec_argv to compare executable name. Thanks!